source/base/Text.ooc

```diff
@@ -135,10 +135,15 @@ Text: cover {
 		this _buffer free(criteria)
 	}
 	slice: func (start, distance: Int) -> This {
-		result := This new(this _buffer slice(start, distance))
-		if (this _buffer owner == Owner Receiver)
-			result = result copy() // TODO: Could we be smarter here?
-		this free(Owner Receiver)
+		result: This
+		if (start == 0 && distance == this count)
+			result = this
+		else {
```

Comment:
@simonmika, since the discussion about saving lines vs performance came up in this PR, why don't we (generally) do something like this?
```ooc
result := this
if (start != 0 || distance != this count) {
    result = This new(this _buffer slice(start, distance))
    if (this _buffer owner == Owner Receiver)
        result = result copy() // TODO: Could we be smarter here?
    this free(Owner Receiver)
}
result
```
It saves 2 lines, and the first assignment, while admittedly unnecessary, has *got* to be cheaper than a function call caused by accessing `this count`.